maxun-core/src/utils/worker-pool.ts

```
@@ -0,0 +1,282 @@
+                        this.updateWorkerMetrics(config.workerIndex, { status: 'failed' });
+                        reject(error);
+                    });
+
+                    worker.on('exit', (code) => {
+                        if (code === 0) {
+                            resolve(workerResults);
+                        } else {
+                            reject(new Error(`Worker stopped with exit code ${code}`));
+                        }
+                    });
+                }).catch(error => {
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)